docs/building-blocks.md

```diff
@@ -0,0 +1,224 @@
+We can take advantage of that to create more intricate expressions.
+
+For example, we can nest `let` expressions to bind more than one name to a value in an expression:
+
+```
+let greeting = "Hello "
+let language = String::toUppercase "dark"
+greeting ++ language
+```
+
+Breaking this down, we can see the sub-expressions that complete the first `let`:
+
+![Nested Let](assets/buildingblocks/nested_let_1.png)
+
+The second `let` (which is the body of the first let) also contains subexpressions:
+
+![Nested Let](assets/buildingblocks/nested_let_2.png)
+
+The value expression of this `let` is a function call containing an expression as its argument:
+
+![Nested Let](assets/buildingblocks/nested_let_3.png)
+
+The body of the second `let` is a string concatenation expression:
+
+![Nested Let](assets/buildingblocks/nested_let_4.png)
+
+The rest of this section covers the kinds of expressions that are part of the Dark language.
+
+## Basic Expressions
+
+### Blanks (Placeholders)
+A blank expressions acts as a placeholder where you can type to create a different expression. The *value* of a blank expression is `Incomplete`, because it needs to be completed in order to be useful.
+
+![Unfilled Blank Example](assets/buildingblocks/blank/blank_example.png)
+
+Once a blank is filled in, it won't be a blank anymore:
+
+![Filled Blank Example](assets/buildingblocks/blank/blank_filled_example.png)
+
+Filling a blank with an expression with sub-parts introduces a new blank for each of them:
+
+![A let with 3 blanks](assets/buildingblocks/blank/blanks_in_let_example.png)
+
+### Partials (Partially Completed Expressions)
+While you are typing or deleting, you'll often see partially completed expressions like this:
+
+![Example of a partial](assets/buildingblocks/partial/partial_example.png)
+
+The text in red is the partial, and the grey text behind it indicates what was there before. Note that partials evaluate to whatever was there before:
+
+![An almost-filled partial still evaluates to what was there before](assets/buildingblocks/partial/partial_almost_filled.png)
+
+...until you complete them from the autocomplete:
+
+![A completed partial is no longer a partial](assets/buildingblocks/partial/partial_completed_example.png)
+
+### Common Atomic Expressions
+Dark supports:
+- Integers (whole numbers like `42`)
+- Floats (numbers like `1.5`, `.51`, and `3.`, which have a whole and fractional part)
+- Strings (text in double quotes like `"Dark is expression-oriented"`)
+- Bools (`true` and `false`)
+- `null`
+
+Each of these expressions evaluates to its contents as you might expect. `42` evaluates to the Int value `42`. Note that numbers in Dark don't have infinite precision. The expression `9999999999999999` evaluates to the Int value `10000000000000000`, as it does in Javascript.
+
+**TODO: Change this because `9999999999999999` behaves differently on frontend and backend!**
+
```

Comment:
I'm not sure what we should change this to, but I relly want something that clarifies the distinction between expressions and their values.

Reply:
I would drop the infinite precision thing. We should fix that, but it's confusing atm so skip it I'd say.